tests/testthat/test-getNeatOrder.R

```diff
@@ -0,0 +1,203 @@
+testthat::set_max_fails(Inf)
+
+
+context("neatsort")
+neatsort_matrix <- matrix(c(10, 8,  2, 8,  3, 
+                            4, 4,  4, 6,  5, 
+                            5, 4,  4, 1,  2, 
+                            2, 3,  5, 7,  8), nrow = 4, ncol = 5, byrow = TRUE, 
+                          dimnames = list(c("Sample1", "Sample2", "Sample3", "Sample4"),
+                                          c("PC1", "PC2", "PC3", "PC4", "PC5")))
+
+test_that("Test getNeatOrder function", {
+    # Test with valid inputs, no subset
+    result <- miaViz::getNeatOrder(neatsort_matrix, dimensions = c(1, 2), centering_method = "mean")
+    expected <- c("Sample4", "Sample2", "Sample3", "Sample1")
+    expect_equal(result, expected)
+    
+    # Test with valid inputs, no subset, decreasing parameter = TRUE
+    result <- miaViz::getNeatOrder(neatsort_matrix, dimensions = c(1, 2), centering_method = "mean", decreasing = TRUE)
+    expected <- c("Sample1", "Sample3", "Sample2", "Sample4")
+    expect_equal(result, expected)
+    
+    # Test with dimensions col names
+    result <- miaViz::getNeatOrder(neatsort_matrix, dimensions = c("PC1", "PC2"), centering_method = "mean")
+    expected <- c("Sample4", "Sample2", "Sample3", "Sample1")
+    expect_equal(result, expected)
+    
+    # Test with valid inputs and subset row names
+    subset <- c("Sample1", "Sample3")
+    result <- miaViz::getNeatOrder(neatsort_matrix, subset, dimensions = c(1, 2), centering_method = "mean")
+    expected <- c("Sample3", "Sample1")
+    expect_equal(result, expected)
+    
+    # Test with valid inputs and subset row indices
+    subset <- c(1, 3)
+    result <- miaViz::getNeatOrder(neatsort_matrix, subset, dimensions = c(1, 2), centering_method = "mean")
```

Comment:
Call functions directly without specifying the package

miaViz::getNeatOrder --> getNeatOrder